extension/goext/core.go

```diff
@@ -0,0 +1,20 @@
+// Copyright (C) 2015 NTT Innovation Institute, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package goext
+
+type CoreInterface interface {
+	RegisterEventHandler(eventName string, handler func(context Context, resource Resource), priority Priority)
```

Comment:
handlers should be allow to modify both context and resource, so *Context and *Resource.

also, handlers should be allowed to report errors (so the return type should be "error")